core/src/main/scala/zio/config/ConfigDescriptorModule.scala

```diff
@@ -998,6 +998,41 @@ trait ConfigDescriptorModule extends ConfigSourceModule { module =>
       loop(self)
     }
 
+    lazy val sources: Set[ConfigSource] = {
```

Comment:
Sources is yet to be used.

Reply:
Most of the logic is just to check if object identity works for recursive structure